test/columns-api.js

```diff
@@ -17,6 +17,35 @@ document.addEventListener('WebComponentsReady', () => {
       });
     });
 
+    describe('manually passed columns', () => {
+      const columns = [
+        {
+          name: 'first',
+          header: 'First Name',
+          path: 'first'
+        },
+        {
+          name: 'last',
+          header: 'Last Name',
+          path: 'last',
+          type: 'custom'
+        },
+        {
+          header: 'Hidden column',
+          hidden: true,
+          path: ''
+        }
+      ];
+
+      it('should set column.id properly', () => {
+        grid.columns = columns;
+        flushVaadinGrid(grid);
+        expect(grid.columns[0].id).to.be.eq('first[string]');
+        expect(grid.columns[1].id).to.be.eq('last[custom]');
```

Comment:
what `id` will be assigned to the hidden column?

Reply:
@limonte, '[string]'. Added test.